### PR TITLE
DOCSP-13865 distinct usage examples

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info

Added Usage Examples > Retrieve Distinct Values of a Field

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13865

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13865-DistinctUsageExample/usage-examples/distinct/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/usage-examples/distinct.txt [hunk 1]
```diff
@@ -3,3 +3,45 @@ Retrieve Distinct Values of a Field
...
+
+Click `here <{+example+}/distinct.go>`__ to see a fully runnable example.
+
+Expected Result
+---------------
+
+After running the preceding code snippet, the ``results`` variable
+should contain the following: 
+
+.. code-block:: json
```
Comment (rustagir) [inline source/usage-examples/distinct.txt:32]: I think this can be a none language

**Review by rustagir — COMMENTED** [review]
Not blocking merging bc all comments are suggestions! Looks really good.

**Comment by kyuan-mongodb** [comment]
Hi @kevinAlbs, This PR is ready for review.

File: source/includes/usage-examples/code-snippets/distinct.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	var uri string
```
Comment (kevinAlbs) [inline source/includes/usage-examples/code-snippets/distinct.go:15]: `uri` should be a const above main, per the [Go Driver Code Example Patterns and Style Reference](https://docs.google.com/document/d/1FtUKxOaEmdFf4d3HcXAVIGed2awuwIWdkWc0YQIwPvk/edit#heading=h.afeu4uin7cta) document.

File: source/includes/usage-examples/code-snippets/distinct.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	// begin distinct
+	coll := client.Database("sample_mflix").Collection("movies")
+	query := bson.D{{"directors", "Natalie Portman"}}
```
Comment (kevinAlbs) [inline source/includes/usage-examples/code-snippets/distinct.go:32]: The [Go driver API docs](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Collection.Distinct) describes this as a filter

```suggestion
	filter := bson.D{{"directors", "Natalie Portman"}}
```

File: source/includes/usage-examples/code-snippets/distinct.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+	results, err := coll.Distinct(context.TODO(), "title", query)
```
Comment (kevinAlbs) [inline source/includes/usage-examples/code-snippets/distinct.go:34]:
```suggestion
	results, err := coll.Distinct(context.TODO(), "title", filter)
```

File: source/usage-examples/distinct.txt [hunk 5]
```diff
@@ -3,3 +3,45 @@ Retrieve Distinct Values of a Field
...
+.. include:: /includes/usage-examples/run-example-tip.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/distinct.go
+   :start-after: begin distinct
+   :end-before: end distinct
+   :emphasize-lines: 4
+   :language: go
+   :dedent:
+
+Click `here <{+example+}/distinct.go>`__ to see a fully runnable example.
```
Comment (kevinAlbs) [inline source/usage-examples/distinct.txt:24]: On the staging link this is currently a 404. Will this be updated later?

File: source/usage-examples/distinct.txt [hunk 6]
```diff
@@ -3,3 +3,45 @@ Retrieve Distinct Values of a Field
...
+
+Expected Result
+---------------
+
+The example returns an ``interface`` type that contains the following
```
Comment (kevinAlbs) [inline source/usage-examples/distinct.txt:29]:
```suggestion
The example returns a slice of empty ``interface`` type that contains the following
```

File: source/usage-examples/distinct.txt (hunk 5 above)
Comment (kyuan-mongodb) [reply source/usage-examples/distinct.txt:24]: yes! once merged into the repo, the link will work

File: source/includes/usage-examples/code-snippets/distinct.go (hunk 2 above)
Comment (kevinAlbs) [reply source/includes/usage-examples/code-snippets/distinct.go:15]: Thank you for updating the patterns doc. I now see the pattern is to use the `MONGODB_URI` environment variable.

**Review by kevinAlbs — APPROVED** [review]
LGTM

**Pull request merged**